src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -206,6 +210,20 @@ private void ProcessNewHeads()
                     {
                         try
                         {
+                            AddressAsKey[]? accountChanges = args.Block.AccountChanges;
+                            if (accountChanges is null || accountChanges.Length == 0 || args.Block.ParentHash != _lastBlockHash || _lastBlockNumber + 1 != args.Block.Number)
+                            {
+                                // Not sequential block, reset cache
+                                _accountCache.Reset();
+                            }
+                            else
+                            {
+                                // Sequential block, just remove changed accounts from cache
+                                _accountCache.RemoveAccounts(accountChanges);
+                            }
+                            _lastBlockNumber = args.Block.Number;
+                            _lastBlockHash = args.Block.Hash;
```

Comment:
IMO: we should extract also changes for those accounts, not only the accounts themselves.

Reply:
And apply the changes to cache? See above